docs/source/tutorials_source/package/tutorial_pretrain_detectron2.py

```diff
@@ -0,0 +1,254 @@
+- How to store the backbone in a checkpoint file which can be used by Detectron2
+
+Prerequisites:
+----------------
+For the purpose of this tutorial you will need to:
+
+- Install Lightly: Follow the `instructions <https://docs.lightly.ai/getting_started/install.html>`_.
+- Install Detectron2: Follow the `instructions <https://detectron2.readthedocs.io/en/latest/tutorials/install.html>`_.
+- Download a dataset for pre-training (we will use the `SKU-110K <https://github.com/eg4000/SKU110K_CVPR19>`_ dataset).
+
+.. note::
+
+    The SKU-110K dataset is an object detection benchmark for densely packed scenes. It consists of more than 10'000
+    images showing retail shelves from varying angles and different camera models. Since contrastive learning works
+    especially well on object-centric images, cropping the objects prior to pre-training would be beneficial. However,
+    we will use the original images for simplicity. Take a look at `lightly-crop` (:ref:`lightly-command-line-tool`)
+    for more info on how to easily extract bounding box crops from your dataset.
+
+Finally, you will need the Detectron2 configuration files. They are available `here <https://github.com/facebookresearch/detectron2/tree/main/configs>`_.
+In this tutorial we use a Faster RCNN with a feature pyramid network (FPN), so make sure you have the relevant file (Base-RCNN-FPN.yaml) in your directory.
+
+
+"""
+
+# %%
+# Imports
+# -------
+#
+# Import the Python frameworks we need for this tutorial.
+import torch
+import lightly
+from detectron2 import config, modeling
+from detectron2.checkpoint import DetectionCheckpointer
+
+# %%
+# Configuration
+# -------------
+# Let's set the configuration parameters for our experiments.
+#
+# We use a batch size of 64 and an input size of 128 in order to fit everything
+# on the available amount of memory on our GPU (16GB). The number of features
+# is set to the default output size of the ResNet50 backbone.
+#
+# We only train for 5 epochs because the focus of this tutorial is on the
+# integration with Detectron2.
+
+num_workers = 8
+batch_size = 64
+input_size = 128
+num_ftrs = 2048
+lr = 0.001
+
+seed = 1
+max_epochs = 5
+
+# use cuda if possible
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+# some images in the dataset are truncated, use this to prevent errors
+from PIL import ImageFile
+ImageFile.LOAD_TRUNCATED_IMAGES = True
+
+# %%
+# You might have downloaded the dataset somewhere else or are using a different one.
+# Set the path to the dataset accordingly. Additionally, make sure to set the
+# path to the config file of the Detectron2 model you want to use.
+# We will be using an RCNN with a feature pyramid network (FPN).
+data_path = '/datasets/SKU110K_fixed/images'
+cfg_path = './Base-RCNN-FPN.yaml'
+
+# %%
+# Initialize the Detectron2 Model
+# --------------------------------
+# 
+# Since the output of the Detectron2 ResNet50 backbone is a dictionary with the keys
```

Comment:
It's not clear why we need this layer and which features we will pick. Maybe we can give a reference to the detectron2 docs? Or maybe we can mention that the different keys `res1` to `res5` correspond to the different stages of the model?
Also mention here already that we only care about the last stage `res5` in this example.